src/gsMSplines/gsWeightMapper.h

```diff
@@ -0,0 +1,486 @@
+/** @file gsWeightMapper.h
+
+    @brief Provides declaration of gsWeightMapper class.
+
+    This file is part of the G+Smo library.
+
+    This Source Code Form is subject to the terms of the Mozilla Public
+    License, v. 2.0. If a copy of the MPL was not distributed with this
+    file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+    Author(s): A. Bressan, F. Buchegger
+*/
+
+#pragma once
+
+#include <gsCore/gsExport.h>
+#include <gsCore/gsLinearAlgebra.h>
+
+namespace gismo
+{
+
+class GISMO_EXPORT gsWeightMapper
```

Comment:
please add 
template<typename T>

Then for templated clases GISMO_EXPORT should be removed.